sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/ExpressionSQLBuilderSuite.scala

```diff
@@ -95,7 +95,7 @@ class ExpressionSQLBuilderSuite extends SparkFunSuite {
 
   test("attributes") {
     checkSQL($"a".int, "a")
-    checkSQL(Symbol("foo bar").int, "`foo bar`")
+    checkSQL(Symbol("`foo bar`").int, "`foo bar`")
```

Comment:
`Symbol("foo bar").int` previously just passed `"foo bar"` to the `AttributeReference` without parsing an `UnresolvedAttribute`. We now call `attr()` which _is_ parsing `"foo bar"` and that fails without backticks.

Alternatively, we can `quoteIfNeeded` in `DslSymbol` and treat, e.g., `Symbol("foo.bar")` as single-part (which is the current behaviour). But I think it's confusing if `$"foo.bar"` returns a multi-part attribute and `Symbol("foo.bar")` doesn't.